x-pack/filebeat/input/http_endpoint/gzip.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package http_endpoint
+
+import (
+	"compress/gzip"
+	"io"
+	"sync"
+)
+
+var gzipDecoderPool = sync.Pool{
+	New: func() interface{} {
+		return new(gzip.Reader)
+	},
+}
+
+type pooledGzipReader struct {
+	Reader *gzip.Reader
+}
+
+func newPooledGzipReader(r io.Reader) (*pooledGzipReader, error) {
+	gzipReader := gzipDecoderPool.Get().(*gzip.Reader) //nolint:errcheck // Only *gzip.Reader are contained in the pool.
```

Comment:
That errcheck complains about this is broken; a runtime panic is a perfectly acceptable check.